rplugin/python3/deoplete/sources/LanguageClientSource.py

```diff
@@ -15,8 +15,9 @@
 
 
 def simplify_snippet(snip: str) -> str:
-    return re.sub(r'(?<!\\)\$\d+', '', snip)
```

Comment:
@HerringtonDarkholme Could you clarify a little bit of this snippet format here? Is it for neosnippet or some other formats?

Reply:
The pattern is from LSP spec. https://github.com/Microsoft/language-server-protocol/blob/cb75aa414c16b2fb1236881570499102b8941080/snippetSyntax.md

I only substitute patterns like `$1`, `$2` which one single regular expression can handle.

Reply:
Thanks. It seems this change covers more cases and should be good to merge in.

Reply:
I wonder how it works if users don't have neosnippet installed. Will deoplete remove snippet placeholder? @yshui

Reply:
I don't think it would. That's why I think we should make this configurable.